### PR TITLE
CurrentEventsByPersistenceId: Stop at the current highest seq nr

### PR DESCRIPTION
**Pull request by chbatey**
Refs #113

### CODE REVIEW

File: core/src/main/scala/akka/persistence/couchbase/scaladsl/CouchbaseReadJournal.scala [hunk 1]
```diff
@@ -286,7 +299,7 @@ final class CouchbaseReadJournal(eas: ExtendedActorSystem, config: Config, confi
             // offset delay ms back in time from now
             TimeBasedUUIDs.create(
               UUIDTimestamp.fromUnixTimestamp(
-                System.currentTimeMillis() - (settings.eventByTagSettings.eventualConsistencyDelay.toMillis)
+                System.currentTimeMillis() - settings.eventByTagSettings.eventualConsistencyDelay.toMillis
```
Comment (johanandren) [inline core/src/main/scala/akka/persistence/couchbase/scaladsl/CouchbaseReadJournal.scala]: heh, that was weird. Did I put those there?

File: core/src/test/scala/akka/persistence/couchbase/scaladsl/EventsByPersistenceIdSpec.scala [hunk 2]
```diff
@@ -123,47 +123,52 @@ class EventsByPersistenceIdSpec extends AbstractQuerySpec("EventsByPersistenceId
...
           .runWith(Sink.seq)
           .futureValue
-        evts.map(_.event) should ===(List(s"$pid-$seqNrAfterInitial"))
+        events.map(_.event) should ===(List(s"$pid-$seqNrAfterInitial"))
       }
 
       // here comes the weird part, the query stage should fill up one page and not do another query until
       // we reach the end of it, but then that second query should see the new event and fail this test
       // but it doesn't
+      val expectedEvents = ((notTheEntireFirstPage + 1L) to initialPersistedEvents).map(n => s"$pid-$n")
```
Comment (johanandren) [inline core/src/test/scala/akka/persistence/couchbase/scaladsl/EventsByPersistenceIdSpec.scala]: the comment is not valid any more

File: core/src/main/scala/akka/persistence/couchbase/scaladsl/CouchbaseReadJournal.scala [hunk 3]
```diff
@@ -155,6 +151,9 @@ final class CouchbaseReadJournal(eas: ExtendedActorSystem, config: Config, confi
                                      toSequenceNr: Long): Source[EventEnvelope, NotUsed] =
     internalEventsByPersistenceId(live = true, persistenceId, fromSequenceNr, toSequenceNr)
 
+  // Should this be configurable?
+  private val queryConsistency = N1qlParams.build().consistency(ScanConsistency.REQUEST_PLUS)
```
Comment (johanandren) [inline core/src/main/scala/akka/persistence/couchbase/scaladsl/CouchbaseReadJournal.scala]: I don't think this shouldn't be the default here, it's expensive, and only really needed on the journal side for replay (see all tests counting on eventual consistency for the index).

File: core/src/main/scala/akka/persistence/couchbase/scaladsl/CouchbaseReadJournal.scala (hunk 1 above)
Comment (chbatey) [reply core/src/main/scala/akka/persistence/couchbase/scaladsl/CouchbaseReadJournal.scala]: Think an elf did

File: core/src/test/scala/akka/persistence/couchbase/scaladsl/EventsByPersistenceIdSpec.scala (hunk 2 above)
Comment (chbatey) [reply core/src/test/scala/akka/persistence/couchbase/scaladsl/EventsByPersistenceIdSpec.scala]: :+1:

File: core/src/main/scala/akka/persistence/couchbase/scaladsl/CouchbaseReadJournal.scala (hunk 3 above)
Comment (chbatey) [reply core/src/main/scala/akka/persistence/couchbase/scaladsl/CouchbaseReadJournal.scala]: :+1: current has no real guarantees

**Comment by johanandren** [comment]
Two failures, one is unrelated, but the other one seems relevant: 
`EventsByPersistenceId must find existing events`

**Comment by chbatey** [comment]
Tests are failing because the query to get the sequenceNr is only executed once

**Comment by johanandren** [comment]
I can take this the last few steps to get it in another RC today.

**Pull request closed without merging**